src/tce/ccsd/trunc_precision.c

```diff
@@ -0,0 +1,17 @@
+#include <x86intrin.h>
+
+// Truncate float32->float16, then convert back to float32 and store in original array memory                         
+void truncate_sp2hp_(float* s_arr, int* sizeP){
```

Comment:
FYI the Intel compiler should auto-vectorize conversions based on bitshift:
https://github.com/jeffhammond/HPCInfo/blob/master/simd/bfloat16.cc#L77

Reply:
Great. I wasn't sure if we'd lose some rounding control with bitshift, so I just threw something together without checking.

Reply:
Err, sorry, `>>` and `<<` are used to convert to/from Bfloat16, not IEEE float16.  For the latter, you need the proper convert instructions.